# Theory Question
Pietro Pizzoccheri 10797420 [team leader]  
Lorenzo Bardelli 10831941

<img src="image.png" alt="Exercise">

### Given Data

In [ ]:
packet_size = 2000 # bits
total_distance = 40 # meters
comm_energy = 300 # nJ/bit
amplifier_energy = 1 # nJ/bit/m^2

## Q1: total communication energy computation

### direct trasmission

In [ ]:
# trasmission energy
E_tx = packet_size * comm_energy + packet_size * amplifier_energy * total_distance**2 # nJ
E_tx_mJ = E_tx / 1e6 # mJ
print(f"Energy for direct transmission: {E_tx_mJ} mJ")

# reception energy
E_rx = packet_size * comm_energy # nJ
E_rx_mJ = E_rx / 1e6 # mJ
print(f"Energy for reception: {E_rx_mJ} mJ")

# total energy
E_total = E_tx + E_rx # nJ
E_total_mJ = E_total / 1e6 # mJ
print(f"Total energy for direct transmission: {E_total_mJ} mJ")

Energy for direct transmission: 3.8 mJ
Energy for reception: 0.6 mJ
Total energy for direct transmission: 4.4 mJ


### multi hop trasmission

In [ ]:
# per hop trasmission energy (assuming uniformly spaced)
E_tx_hop = packet_size * comm_energy + packet_size * amplifier_energy * (total_distance/4)**2 # nJ
E_tx_hop_mJ = E_tx_hop / 1e6 # mJ
print(f"Energy for transmission per hop: {E_tx_hop_mJ} mJ")

# per hop reception energy
E_rx_hop = packet_size * comm_energy # nJ
E_rx_hop_mJ = E_rx_hop / 1e6 # mJ
print(f"Energy for reception per hop: {E_rx_hop_mJ} mJ")

# total energy per hop
E_hop = E_tx_hop + E_rx_hop # nJ
E_hop_mJ = E_hop / 1e6 # mJ
print(f"Total energy per hop: {E_hop_mJ} mJ")

# total energy for multi hop trasmission
E_total_multi_hop = E_hop * 4 # nJ
E_total_multi_hop_mJ = E_total_multi_hop / 1e6 # mJ
print(f"Total energy for multi hop transmission: {E_total_multi_hop_mJ} mJ")

Energy for transmission per hop: 0.8 mJ
Energy for reception per hop: 0.6 mJ
Total energy per hop: 1.4 mJ
Total energy for multi hop transmission: 5.6 mJ


## Q2: Questions

### Which strategy would you choose for long-term operation?

We would choose the multi hop trasmission strategy since it is more energy efficient and allows to extend the battery life of the sensor. 
The direct trasmission strategy may be simpler to implement but it would drain the battery much faster and would require more frequent battery replacements, which is not ideal for long-term operation.



### Which performance metric are you optimizing when choosing the strategy for long-term operation?

The performance metric we are optimizing is the battery life of the sensor (node lifetime of the most energy costrained node).
By choosing the multi hop trasmission strategy, we are optimizing for a longer battery life, which is crucial for long-term operation and reduces the need for frequent maintenance and battery replacements.



### Provide a brief justification. Your explanation must clearly link the selected strategy to the chosen performance metric and be logically consistent with your numerical results.

While the numerical results from Q1 show that the total network energy is higher for multi-hop (5.6 mJ) compared to direct transmission (4.4 mJ), looking at the manufacturer specifications reveals a critical hardware constraint:

- The Aqara Motion Sensor is powered by a limited battery (2 × CR2450).

- The Philips Hue smart bulbs (B1, B2, B3) are connected directly to the electrical grid.

In direct transmission, the battery-powered sensor consumes 3.8 mJ per message to reach the sink. 
In multi-hop transmission, the sensor only consumes 0.8 mJ to reach the first bulb (B1)—nearly a 5x reduction in power draw for the sensor.

By choosing multi-hop, we intentionally shift the energy consumption away from the constrained battery node onto the powered bulbs. This drastically extends the time before human intervention is required to replace the sensor's batteries, thereby maximizing the long-term autonomous operation of the entire system.